### ISSUE TITLE
Showing relative destination to drivers for transparency

### ISSUE THREAD

**Issue by ashwingattani** [p2prider/p2prider#17]
Similar to Airbnb, when the driver gets a new ride request, they would be shown a relative destination area for better decision making. Something like this,
<img width="1141" alt="Screenshot 2022-03-21 at 3 54 57 PM" src="https://user-images.githubusercontent.com/9816850/159248601-93f90dfe-84a3-4549-95e4-49fcd6c679f6.png">

Need more discussion on this.

**Comment by p2prider** [comment]
I think this is an interesting idea - to mask the drop location, from a privacy perspective. Needs more thought on that regard, however.

> they would be shown a relative destination area for better decision making

Hmm, perhaps I am missing something obvious but how does it help in better decision making for the driver? Could you give an example?

**Comment by ashwingattani** [comment]
> Hmm, perhaps I am missing something obvious but how does it help in better decision making for the driver? Could you give an example?

A pre-known destination would help drivers accept or cancel rides right away instead of calling the customer and then canceling it. It would also helo drivers analyze whether the rate provided is justifiable or not.

**Comment by p2prider** [comment]
> A pre-known destination would help drivers accept or cancel rides right away instead of calling the customer and then canceling it. It would also helo drivers analyze whether the rate provided is justifiable or not.

Sure, but it's the same (actually better) when you show the exact drop location and the route to it. My question is, how does showing a masked/rough location instead of the exact user-chosen drop location help either party? (other than the privacy thing, though I think that's a different debate which we should have if that is the main point)

**Comment by ashwingattani** [comment]
Privacy was the primary concern for me. But now as I think of it, I am not sure if showing the exact location would hamper privacy. The question is, why aren't Ola and Uber not showing the destination then?

Showing the complete route before the ride can be complex in the app development, I am not sure if we should do that. The drivers anyhow know the best possible route to any destination in the city, and we can show gmaps when the ride starts.

**Comment by p2prider** [comment]
> Privacy was the primary concern for me. But now as I think of it, I am not sure if showing the exact location would hamper privacy.

Indeed. Couple of points regarding this (which I think you already know, but for other readers/posterity):

1. This runs counter to our theme of transparency for everyone involved.
2. On our platform, the driver can workaround this extremely easily by accepting the trip, checking the precise location and then cancelling. Baking in disincentives for this behavior would be difficult. And this behavior will be frustrating and confusing for the riders.
3. Users can choose a rough drop location on their own depending on how much privacy they want.
4. Drivers only see a number for the user, no names anyway.

> The question is, why aren't Ola and Uber not showing the destination then?

I think their goal is to not let the drivers have any say in whether they should accept the trip or not.

> Showing the complete route before the ride can be complex in the app development, I am not sure if we should do that.

Yes. Which is why the original suggestion was:

>       ii. A button to "preview the route".
>           a. NOTE: Actual behavior of this button might vary depending on availability of in-app route preview or deeplink to google maps, but it will be the same button regardless.

That is, we don't load the route for every trip. Only if the driver is interested in the trip from the basic info shown, he can click on the preview button to get more info. Route generation would be done optimistically, and fallback would always be to deeplink to google maps.

> The drivers anyhow know the best possible route to any destination in the city, and we can show gmaps when the ride starts.

I largely agree. But we should take user feedback on this and verify this.

**Comment by papasierragolf** [comment]
> 

Our goal is avoid the issues which current ride aggregator users face. 
To give the driver the transparency to accept / decline ride after seeing the trip preview. This helps to prevent unnecessary cancellations for the rider / driver. 

On the privacy aspect, the driver will be getting to know where they are dropping the user. Its the same for non digital services like a traditional auto / cab hailing. The rider will tell the driver where they want to go. Depending on their need for privacy, they can select the destination

And while booking the trip, we can tell the driver 3 things

1. distance from their location to pickup point.
2. distance of trip
3. preferences of the rider if any - such as all time AC, etc 


Also regarding route generation, it is better to leave it on google maps / apple maps (depending on the area / country) because building a whole new system would inherently be a waste of time since Google's solution improves everyday.
